File: LinkedList/Medium/02-Reverse a LinkedList [Iterative].java
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null || head.next == null){
            return head;
        }

        ListNode prevNode = null;
        ListNode currNode = head;
        ListNode nextNode = head.next;

        while(currNode != null){
            currNode.next = prevNode;
            prevNode = currNode;
            currNode = nextNode;
            if(nextNode != null){
                nextNode = nextNode.next;
            }
        }

        return prevNode;
    }
}

File: LinkedList/Medium/02-Reverse a LinkedList [Iterative].cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head == nullptr || head -> next == nullptr){
            return head;
        }

        ListNode* prevNode = nullptr;
        ListNode* currNode = head;
        ListNode* nextNode = head -> next;

        while(currNode){
            currNode -> next = prevNode;
            prevNode = currNode;
            currNode = nextNode;

            if(nextNode){
                nextNode = nextNode -> next;
            }
        }

        return prevNode;
    }
};